src/odemis/driver/technolution.py

```diff
@@ -604,10 +591,15 @@ def __init__(self, name, role, parent, **kwargs):
         # direction of the executed scan
         self.rotation = model.FloatContinuous(0.0, range=(0.0, 2 * math.pi), unit='rad')
 
+        # Scanner settings:
+        # In x we typically scan from negative to positive centered around zero and
+        # in y from positive to negative centered around zero.
         # the start of the sawtooth scanning signal
-        self.scanOffset = model.TupleContinuous((0.0, 0.0), range=((-1.0, -1.0), (1.0, 1.0)))
-        # the end (amplitude) of the sawtooth scanning signal
-        self.scanGain = model.TupleContinuous((0.3, 0.3), range=((-1.0, -1.0), (1.0, 1.0)))
+        self.scanOffset = model.TupleContinuous((0.0, 0.0), range=((-1.0, -1.0), (1.0, 1.0)), cls=(int, float))
+        # heights of the sawtooth scanner signal (it does not include the offset!)
+        self.scanAmplitude = model.TupleContinuous((0.1, -0.1), range=((-1.0, -1.0), (1.0, 1.0)), cls=(int, float))
+        # FIXME add a check that offset + amplitude >! 2**15 - 1 and offset + amplitude <! -2**15
```

Comment:
Preferably, implement it now. Or what's preventing you to do so? Maybe it's a little tricky to raise error when 2 VAs combined are incompatible. As seen before, it might be better to automatically adjust one of them instead of raising an error.
An alternative would be to just log a warning (though it's a lot less effective to notice that something is wrong).

Reply:
I didn't do it because of the time that was originally planned for this task. And it took me already way longer. So far this check was also not there and would have also raised an error when acquiring an image. That's why I thought I only put the FIXME and do it in a separate pr. 
I agree, a warning is not useful as we only would notice when the acquisition is failing.
So we could do as you suggested:

1. raise an error (which informs the user what value range is possible with the current value of the other VA) and return the current value (which would be the easiest implementation)
2. return the max/min value that is still possible (but if offset = 1, and the amplitude = -0.5 -> what do you return if the user requests amplitude = 1? `-1 or 0`?) This option needs some more thought I think.

What do you think?

Reply:
I will leave this for a separate pr.